workflow-to-js-transformer/src/main/java/com/github/cafdataprocessing/workflow/transform/FullWorkflowRetriever.java

```diff
@@ -178,4 +227,38 @@ private FullAction buildFullAction(
         }
         return new FullAction(existingAction, actionConditions);
     }
+
+    private Long getWorkflowId(final String projectId, final WorkflowsApi workflowsApi, final String workflowName)
+        throws ApiException, InvalidWorkflowSpecificationException
+    {
+        final ExistingWorkflows existingWorkflows = workflowsApi.getWorkflows(projectId, 1, 100);
+        final Map<String, Long> workflows = existingWorkflows.getWorkflows().stream().collect(
+            Collectors.toMap(ExistingWorkflow::getName, ExistingWorkflow::getId));
+        if (!workflows.containsKey(workflowName)) {
+            throw new InvalidWorkflowSpecificationException("The name of the workflow provided could not be resolved.");
+        }
+        return workflows.get(workflowName);
+    }
+
+    private final class CacheInfoSupplier
```

Comment:
This is used as a cache key, but `equals()` hasn't been implemented.

Reply:
Also this shouldn't be a inner class.  If you want to use a nested class then you need to make it static.